sycl/source/detail/memory_manager.cpp

```diff
@@ -414,13 +436,11 @@ void *MemoryManager::map(SYCLMemObjT *SYCLMemObj, void *Mem, QueueImplPtr Queue,
 
 void MemoryManager::unmap(SYCLMemObjT *SYCLMemObj, void *Mem,
                           QueueImplPtr Queue, void *MappedPtr,
-                          std::vector<cl_event> DepEvents,
-                          bool UseExclusiveQueue, cl_event &OutEvent) {
+                          std::vector<cl_event> DepEvents, cl_event &OutEvent) {
+  
   cl_int Error = CL_SUCCESS;
   Error = clEnqueueUnmapMemObject(
-      UseExclusiveQueue ? Queue->getExclusiveQueueHandleRef()
-                        : Queue->getHandleRef(),
-      (cl_mem)Mem, MappedPtr, DepEvents.size(),
+      Queue->getHandleRef(), (cl_mem)Mem, MappedPtr, DepEvents.size(),
```

Comment:
please keep the ternary statement that checks UseExclusiveQueue and passes different handle refs, unless there is good reason.